internal/services/dataprotection/data_protection_backup_policy_mysql_flexible_server_resource.go

```diff
@@ -0,0 +1,697 @@
+									ValidateFunc: azValidate.ISO8601Duration,
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+
+		"retention_rule": {
+			Type:     pluginsdk.TypeList,
+			Optional: true,
+			ForceNew: true,
+			Elem: &pluginsdk.Resource{
+				Schema: map[string]*pluginsdk.Schema{
+					"name": {
+						Type:         pluginsdk.TypeString,
+						Required:     true,
+						ForceNew:     true,
+						ValidateFunc: validation.StringIsNotEmpty,
+					},
+
+					"criteria": {
+						Type:     pluginsdk.TypeList,
+						Required: true,
+						ForceNew: true,
+						MaxItems: 1,
+						Elem: &pluginsdk.Resource{
+							Schema: map[string]*pluginsdk.Schema{
+								"absolute_criteria": {
+									Type:         pluginsdk.TypeString,
+									Optional:     true,
+									ForceNew:     true,
+									ValidateFunc: validation.StringInSlice(backuppolicies.PossibleValuesForAbsoluteMarker(), false),
+								},
+
+								"days_of_week": {
+									Type:     pluginsdk.TypeSet,
+									Optional: true,
+									ForceNew: true,
+									MinItems: 1,
+									Elem: &pluginsdk.Schema{
+										Type:         pluginsdk.TypeString,
+										ValidateFunc: validation.StringInSlice(backuppolicies.PossibleValuesForDayOfWeek(), false),
+									},
+								},
+
+								"months_of_year": {
+									Type:     pluginsdk.TypeSet,
+									Optional: true,
+									ForceNew: true,
+									MinItems: 1,
+									Elem: &pluginsdk.Schema{
+										Type:         pluginsdk.TypeString,
+										ValidateFunc: validation.StringInSlice(backuppolicies.PossibleValuesForMonth(), false),
+									},
+								},
+
+								"scheduled_backup_times": {
+									Type:     pluginsdk.TypeSet,
+									Optional: true,
+									ForceNew: true,
+									MinItems: 1,
+									Elem: &pluginsdk.Schema{
+										Type:         pluginsdk.TypeString,
+										ValidateFunc: validation.IsRFC3339Time,
+									},
+								},
+
+								"weeks_of_month": {
+									Type:     pluginsdk.TypeSet,
+									Optional: true,
+									ForceNew: true,
+									MinItems: 1,
+									Elem: &pluginsdk.Schema{
+										Type:         pluginsdk.TypeString,
+										ValidateFunc: validation.StringInSlice(backuppolicies.PossibleValuesForWeekNumber(), false),
+									},
+								},
+							},
+						},
+					},
+
+					"life_cycle": {
+						Type:     pluginsdk.TypeList,
+						Required: true,
+						ForceNew: true,
+						Elem: &pluginsdk.Resource{
+							Schema: map[string]*pluginsdk.Schema{
+								"data_store_type": {
+									Type:     pluginsdk.TypeString,
+									Required: true,
+									ForceNew: true,
+									ValidateFunc: validation.StringInSlice([]string{
+										// Confirmed with the service team that currently only `VaultStore` is supported.
+										// However, considering that `ArchiveStore` will be supported in the future, it would be exposed for user specification.
+										string(backuppolicies.DataStoreTypesVaultStore),
+									}, false),
```

Comment:
as there is only 1 possible value, could we make this optional with it as the default?

Reply:
Based on below reasons. I suggest aligning with other the existing data protection backup policies.

When new value is introduced in the future, it may cause breaking change if the default value is changed.
Or when new value is introduced in the future, it may cause breaking change since this property needs to be updated from optional to required if the default value is no longer need.
TF has supported many [data protection backup policies](https://github.com/hashicorp/terraform-provider-azurerm/tree/main/internal/services/dataprotection). They are almost all the same. I feel it should be aligned with others for unifying the style.